src/algebraic_topology/dold_kan/split_simplicial_object.lean

```diff
@@ -109,6 +110,178 @@ begin
     degeneracy_comp_P_infty X n A.e hA, comp_zero],
 end
 
+lemma comp_P_infty_eq_zero_iff {Z : C} {n : ℕ} (f : Z ⟶ X _[n]) :
+  f ≫ P_infty.f n = 0 ↔ f ≫ s.π_summand (index_set.id (op [n])) = 0 :=
+begin
+  split,
+  { intro h,
+    cases n,
+    { dsimp at h,
+      rw [comp_id] at h,
+      rw [h, zero_comp], },
+    { have h' := f ≫= P_infty_f_add_Q_infty_f (n+1),
+      dsimp at h',
+      rw [comp_id, comp_add, h, zero_add] at h',
+      rw [← h', assoc, Q_infty_f, decomposition_Q, preadditive.sum_comp,
+        preadditive.comp_sum, finset.sum_eq_zero],
+      intros i hi,
+      simp only [assoc, σ_comp_π_summand_id_eq_zero, comp_zero], }, },
+  { intro h,
+    rw [← comp_id f, assoc, s.decomposition_id, preadditive.sum_comp,
+      preadditive.comp_sum, fintype.sum_eq_zero],
+    intro A,
+    by_cases hA : A.eq_id,
+    { dsimp at hA,
+      subst hA,
+      rw [assoc, reassoc_of h, zero_comp], },
+    { simp only [assoc, s.ι_summand_comp_P_infty_eq_zero A hA, comp_zero], }, },
+end
+
+@[simp, reassoc]
+lemma P_infty_comp_π_summand_id (n : ℕ) :
+  P_infty.f n ≫ s.π_summand (index_set.id (op [n])) = s.π_summand (index_set.id (op [n])) :=
+begin
+  conv_rhs { rw ← id_comp (s.π_summand _), },
+  symmetry,
+  rw [← sub_eq_zero, ← sub_comp, ← comp_P_infty_eq_zero_iff, sub_comp, id_comp,
+    P_infty_f_idem, sub_self],
+end
+
+@[simp, reassoc]
+lemma π_summand_comp_ι_summand_comp_P_infty_eq_P_infty (n : ℕ) :
+  s.π_summand (index_set.id (op [n])) ≫ s.ι_summand (index_set.id (op [n])) ≫ P_infty.f n =
+    P_infty.f n :=
+begin
+  conv_rhs { rw ← id_comp (P_infty.f n), },
+  erw [s.decomposition_id, preadditive.sum_comp],
+  rw [fintype.sum_eq_single (index_set.id (op [n])), assoc],
+  rintros A (hA : ¬A.eq_id),
+  rw [assoc, s.ι_summand_comp_P_infty_eq_zero A hA, comp_zero],
+end
+
+/-- The differentials `s.d i j : s.N i ⟶ s.N j` on nondegenerate simplices of a split
+simplicial object are induced by the differentials on the alternating face map complex. -/
+@[simp]
+def d (i j : ℕ) : s.N i ⟶ s.N j :=
+s.ι_summand (index_set.id (op [i])) ≫ K[X].d i j ≫ s.π_summand (index_set.id (op [j]))
+
+lemma ι_summand_comp_d_comp_π_summand_eq_zero (j k : ℕ) (A : index_set (op [j])) (hA : ¬A.eq_id) :
+  s.ι_summand A ≫ K[X].d j k ≫ s.π_summand (index_set.id (op [k])) = 0 :=
+begin
+  rw A.eq_id_iff_mono at hA,
+  rw [← assoc, ← s.comp_P_infty_eq_zero_iff, assoc, ← P_infty.comm j k, s.ι_summand_eq, assoc,
+    degeneracy_comp_P_infty_assoc X j A.e hA, zero_comp, comp_zero],
+end
+
+/-- If `s` is a splitting of a simplicial object `X` in a preadditive category, `s.N'` is
+a chain complex which is given in degree `n` by the nondegenerate `n`-simplices of `X`. -/
+@[simps]
+def N' : chain_complex C ℕ :=
```

Comment:
This construction is generally useful outside of Dold-Kan, right? It's not just an implementation detail, I think. So maybe it's is good to give it a slightly more descriptive name? Maybe something like `nondeg_complex` or something with `nondegenerate` in the name.

What do you think?